### PR TITLE
Bump minimal PHP version to ^5.5.9|>=7.0.8

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #22973
| License       | MIT
| Doc PR        | -

As spotted in the linked issue, because of https://bugs.php.net/72229.

### CODE REVIEW

**Comment by curry684** [comment]
This will just confuse users with <7.0.8 about why they're not getting important security updates anymore.

**Comment by nicolas-grekas** [comment]
@curry684 truth is, serializing is broken with PHP <7.0.8, so everybody should reject these versions. It's only losely related to the linked issue.

**Review by fabpot — APPROVED** [review]

**Review by Simperfit — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Comment by curry684** [comment]
I completely agree with the solution and change, just not some of its possible side effects.

If 3.3.6 contains a warning for PHP <7.0.8, and we then remove that in 3.3.7 and 3.4.0 along with the Composer bump the issue is fixed, *and* users that are upgrading Symfony but not PHP have a clear end-of-road warning.

**Comment by nicolas-grekas** [comment]
@curry684 see #23705

@symfony/deciders should be merged after the release of v3.3.6

**Review by ogizanagi — APPROVED** [review]

**Comment by curry684** [comment]
Cheers, well solved.

File: src/Symfony/Component/Form/Extension/DataCollector/FormDataCollector.php [hunk 1]
```diff
@@ -85,9 +85,6 @@ public function __construct(FormDataExtractorInterface $dataExtractor)
      */
     public function collect(Request $request, Response $response, \Exception $exception = null)
     {
-        if (70000 <= \PHP_VERSION_ID && \PHP_VERSION_ID < 70800) {
-            @trigger_error('A bug in PHP 7.0.0 to 7.0.7 is breaking the Form panel, please upgrade to 7.0.8 or higher.', E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/DataCollector/FormDataCollector.php]: now removed

**Review by javiereguiluz — APPROVED** [review]

**Pull request merged**